app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,59 @@
+class OrdersController < ApplicationController
+
+  before_action :set_order, only: [:show, :cancel, :paid, :completed]
+
+  def index
+    @orders = current_user.orders
+  end
+
+  def show
+    if current_user.nil? || !current_user.orders.include?(@order) && !current_user.admin?
+      flash[:danger] = "You can only view your own orders. Please log in with the appropritate credentials."
+      redirect_to root_path
+    end
+  end
+
+  def create
+    if session[:user_id].nil?
+      redirect_to login_path
+    else
```

Comment:
I suggest a before_action to do the redirect before the create. That would clean up the if statements.